### PR TITLE
Add Console Based Ecommerce App

### PR DESCRIPTION
**Pull request by prem-sagar2000**
This is console based E-commerce app.

### CODE REVIEW

File: ecommerce_app.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
+from user import User
+from product import Product
```
Comment (faatehsultan) [inline ecommerce_app.py:2]: your imports should be ordered

File: ecommerce_app.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+
+class ECommerce:
+    def __init__(self):
+        self.users = []
+        self.current_user = None
+        self.products = []
+    
+    def get_user_input(self, display_text, invalid_message, dtype = str):
+        while True:
```
Comment (faatehsultan) [inline ecommerce_app.py]: avoid using infinit loop and replace with proper conditions

File: ecommerce_app.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+'''
+
+
+class ECommerce:
+    def __init__(self):
+        self.users = []
+        self.current_user = None
+        self.products = []
+    
+    def get_user_input(self, display_text, invalid_message, dtype = str):
```
Comment (faatehsultan) [inline ecommerce_app.py]: shift this function to a file called utils.py as this is a utility function and then import it

File: ecommerce_app.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            if user.email == email:
+                print('User with this email already exists. Please try again.')
+                return
+
+        new_user = User(name, password, email)
+        self.users.append(new_user)
+        print('Registration successful!')
+        
+    def isValidInput(self, inp=None):
+        return inp and inp in range(0,7) or True
```
Comment (faatehsultan) [inline ecommerce_app.py]: you can make the range dynamic

File: ecommerce_app.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        for user in self.users:
+            if user.email == email:
+                print('User with this email already exists. Please try again.')
+                return
+
+        new_user = User(name, password, email)
+        self.users.append(new_user)
+        print('Registration successful!')
+        
+    def isValidInput(self, inp=None):
```
Comment (faatehsultan) [inline ecommerce_app.py]: this function should also be in utils.py

File: ecommerce_app.py [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    def run_main_menu(self):
+        choice = None
+        while self.isValidInput(choice):
+            print(MENU)
+            choice = input('Enter your choice (1-7): ')
+            try:
+                choice = int(choice)
+            except:
+                choice = None
+                continue
```
Comment (faatehsultan) [inline ecommerce_app.py]: avoid using continue and break statements

File: ecommerce_app.py [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+                self.list_products()
+            elif choice == 7:
+                print('Exiting the application.')
+                break
+            else:
+                print('Invalid choice. Please try again.')
+            
+            
+app = ECommerce()
+app.run_main_menu()
```
Comment (faatehsultan) [inline ecommerce_app.py]: add empty line

File: product.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class Product:
+    def __init__(self, name, quantity, price):
+        self.name = name
+        self.quantity = quantity
+        self.price = price
```
Comment (faatehsultan) [inline product.py:5]: add empty line

File: user.py [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class User:
+    def __init__(self, name, password, email):
+        self.name = name
+        self.password = password
+        self.email = email
```
Comment (faatehsultan) [inline user.py:5]: add empty line

**Review by faatehsultan — CHANGES_REQUESTED** [review]

File: product.py (hunk 8 above)
Comment (prem-sagar2000) [reply product.py:5]: There is already an empty line.

File: user.py (hunk 9 above)
Comment (prem-sagar2000) [reply user.py:5]: There is also an empty line at the end of file. i.e. there are 6 lines in this file I don't know why it doesn't appear on your side.

**Review by faatehsultan — APPROVED** [review]

**Pull request merged**